src/DependencyInjection/Compiler/AdminSearchCompilerPass.php

```diff
@@ -41,14 +41,15 @@ public function process(ContainerBuilder $container): void
         foreach ($container->findTaggedServiceIds(TaggedAdminInterface::ADMIN_TAG) as $id => $tags) {
             $this->validateAdminClass($container, $id);
 
+            $adminCode = $container->getDefinition($id)->getArgument(0);
```

Comment:
I don't see a better way to get the admin code...

Reply:
This will be a problem for auto configuration right?

Reply:
This will need to be done in another way the day we'll have full-auto configuration. Maybe later this could be replace by some static `$class::getCode` method call. But we currently don't. Even https://github.com/sonata-project/SonataAdminBundle/pull/7660 since to not touch the constructor definition.

Since the code is used to access the value in the SearchHandler, I need to access to the code in the compiler pass.
In the ExtensionCompilerPass, the call `$admin->getArgument(1)` is made, so this won't be the first time we rely on the argument.

Reply:
When relying on the first argument, we should check if this is a valid `string` or use the `id` as a fallback

Reply:
I added a string check.